### ISSUE TITLE
Wide review tracker

### ISSUE THREAD

**Issue by anssiko** [w3c/compute-pressure#177]
### About

This is a meta issue to track [wide review](https://www.w3.org/2021/Process-20211102/#wide-review) for the [Compute Pressure API](https://www.w3.org/TR/compute-pressure/).

An important part of wide review is horizontal review from W3C's [key horizontal groups](https://www.w3.org/Guide/documentreview/#how_to_get_horizontal_review) listed below in **horizontal groups** section. Also feedback from **other stakeholders** is equally important. Additional pointers are welcome via comments.

The list is based on the [How to do wide review](https://www.w3.org/Guide/documentreview/#how_to_get_horizontal_review) page guidance.

>Legend:
>🔴 Review request not submitted
>🟡 Review request submitted
>🔵 Review feedback received
>🟢 Review closed as completed

### Horizontal groups

🟢 ♿ **Accessibility**

- [x] Work through <a rel="nofollow" class="external text" href="https://w3c.github.io/apa/fast/checklist.html">this questionnaire</a>
   - [x] #183
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/a11y-request/issues/new/choose">request a review via GitHub</a> from <a rel="nofollow" class="external text" href="https://www.w3.org/WAI/APA/">APA</a>
  - [x] https://github.com/w3c/a11y-request/issues/53 
    - [x] Review closed as completed with substantive contributions #247 

🟢 📐 **Architecture**

- [x] Ask the <a rel="nofollow" class="external text" href="https://www.w3.org/2001/tag/">TAG</a> for review; see <a rel="nofollow" class="external text" href="https://tag.w3.org/workmode/">how to work with the TAG</a>
   - [x] https://github.com/w3ctag/design-reviews/issues/795
   - [x] Positive feedback received on data minimization approach: https://github.com/w3ctag/design-reviews/issues/795#issuecomment-1513997968

- [x] If you are developing javascript APIs you may also want to ask <a rel="nofollow" class="external text" href="mailto:public-script-coord@w3.org">public-script-coord@w3.org</a>, a technical discussion list shared by W3C and ECMA's TC 39
   - [x] No issues related to JS language features that should be brought to TC 39 for consideration.

🟢 🌍 **Internationalisation**

- [x] Read the <a rel="nofollow" class="external text" href="https://www.w3.org/International/review-request">Request a review</a> page
- [x] work through the <a rel="nofollow" class="external text" href="https://www.w3.org/International/i18n-drafts/techniques/shortchecklist">Short Checklist</a>
  - [x] #184
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/i18n-request/issues/new/choose">request a review via GitHub</a>
  - [x] https://github.com/w3c/i18n-request/issues/204
    - [x] Review closed as completed https://github.com/w3c/i18n-request/issues/204#issuecomment-1452042009

🟢 🔍 **Privacy**

- [x] Write a "Privacy Considerations" section for your document, taking into account the <a rel="nofollow" class="external text" href="https://www.w3.org/TR/security-privacy-questionnaire/">Self-Review Questionnaire: Security and Privacy</a>, <a rel="nofollow" class="external text" href="https://www.w3.org/TR/fingerprinting-guidance/">Mitigating Browser Fingerprinting in Web Specifications</a>, and <a rel="nofollow" class="external text" href="https://www.rfc-editor.org/rfc/rfc6973.html">RFC6973</a>
   - [x] Completed [Self-Review Questionnaire: Security and Privacy](https://github.com/w3c/compute-pressure/blob/main/security-privacy-self-assessment.md)
   - [x] Authored [Spec: Privacy Considerations](https://www.w3.org/TR/compute-pressure/#security-and-privacy-considerations)
   - [x] Chrome Security and Privacy team feedback https://github.com/w3c/compute-pressure/issues/79
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3cping/privacy-request/issues/new/choose">request a review via GitHub</a> from the <a rel="nofollow" class="external text" href="https://www.w3.org/Privacy/">Privacy Interest Group</a>
  - [x] https://github.com/w3cping/privacy-request/issues/113
    - [x] Review closed as completed with substantive contributions https://github.com/w3cping/privacy-request/issues/113#issuecomment-1477173013 

🟢 🔒 **Security**
- [x] Write a "Security Considerations" section for your document, taking into account the <a rel="nofollow" class="external text" href="https://www.w3.org/TR/security-privacy-questionnaire/">Self-Review Questionnaire: Security and Privacy</a>
   - [x] Authored [Spec: Security Considerations](https://www.w3.org/TR/compute-pressure/#security-and-privacy-considerations)   
   - [x] Chrome Security and Privacy team feedback https://github.com/w3c/compute-pressure/issues/79
- [x] <a rel="nofollow" class="external text" href="https://github.com/w3c/security-request/issues/new/choose">request a review via GitHub</a>
  - [x] https://github.com/w3c/security-request/issues/50
    - [x] Review closed as completed: https://github.com/w3c/security-request/issues/50#issuecomment-1520179686 

### Other stakeholders

From [who to ask for review](https://www.w3.org/Guide/documentreview/#who_to_ask_for_review):
>Horizontal reviews [...] are only a subset of a full wide review, which must also include other stakeholders including Web developers, technology providers and implementers not active in the Working Group, external groups and standards organizations working on related areas, etc.

🟢 Zoom - positive signals and support https://github.com/w3c/compute-pressure/issues/14

**Comment by anssiko** [comment]
I have now initiated the **Privacy** and **Security** reviews with the respective horizontal groups.

In addition, I have staged both the **Accessibility** and **Internationalisation** reviews for submission, pending editors' review of the checklist materials to be provided as references in these requests.

Lastly, we had submitted the **Architecture** (aka TAG) review request earlier.

The status of our overall wide review progress is being tracked in this issue with links to the relevant material and feedback. Our goals is to turn all 🔴 into 🟡 during Q1, and push to 🟢 during Q2.

Your diligent efforts ensure that the entire set of stakeholders of the web community, including the general public, have had adequate notice of the progress of the Working Group and are able to actually perform reviews of and provide comments on the specification.

Our second objective is to make sure we request reviews early enough that comments and suggested changes can still be reasonably incorporated in response to the review.

Thank you for your contributions.

**Comment by anssiko** [comment]
Current status:

- All wide review requests have been submitted.

- Internationalisation review has been completed.

- Privacy review feedback has been received and is being addressed.

Please refer to the tracker in the first comment for pointers.

**Comment by anssiko** [comment]
Current status:

- Accessibility review request queued for processing.

- Architecture review received. Positive feedback with a request to share developer feedback when available. Confirming if further feedback is expected.

- Privacy review feedback broken into [privacy-needs-resolution issues](https://github.com/w3c/compute-pressure/issues?q=is%3Aissue+is%3Aopen+label%3Aprivacy-needs-resolution) welcomes contributions.

- Security review closed as completed.

**Comment by anssiko** [comment]
I'm thrilled to announce wide review for the Compute Pressure API has been completed. 🥳 

The Devices and Sensors Working Group would like to thank numerous contributors across Accessibility, Architecture, Internationalization, Privacy and Security horizontals for their deligent review and contributions.

**Comment by himorin** [comment]
@anssiko Next step toward CR after HR, would be settling down all spec issues (or marking related part as at-risk?), and call for consensus to publish as CR within the group, I believe.
It seems there is only small number of registered issues (except for V2 or enhancement), we may go further shortly. How do you think? (of course, all works might start after new year.)

**Comment by anssiko** [comment]
Feedback from early adopters of the Compute Pressure API is now available: https://github.com/w3ctag/design-reviews/issues/795#issuecomment-1982796332

This feedback suggests the current version of the API is addressing real customer needs. The feedback also motivates possible future work.

@kenchris @arskama please ensure that we have recorded the relevant feedback for possible future work as GH issues in this repo, labeled as https://github.com/w3c/compute-pressure/labels/V2

**Comment by anssiko** [comment]
The Compute Pressure API [shipped in Chrome 125](https://developer.chrome.com/blog/new-in-chrome-125/#compute-pressure-api) Stable release yesterday. Congratulations everyone who made this happen!

This important implementation milestone was reached after an extensive trial period with real-world customers between July 2023 and March 2024. Feedback received during this trial period demonstrated the API addresses important end user needs and motivated the shipping decision.

Considering this significant advancement on the implementation front, I recommend the WG to start advancing this specification toward the Candidate Recommendation stage in the near future. This transition is to formally signal the specification is welcoming further [implementation experience](https://www.w3.org/2023/Process-20231103/#implementation-experience).

We will assess the CR transition readiness in the coming months with assistance from @himorin and will follow up with more information and concrete next steps. Meanwhile, I encourage the editors @kenchris @arskama to triage the remaining open issues and label any issues considered out of scope for the expected CR, including any new features, as https://github.com/w3c/compute-pressure/labels/V2 This is to help delineate what should be addressed by CR from what comes after.

**Comment by pes10k** [comment]
Thats wonderful @anssiko .  I'm repeating myself I know, but i just wanted to appreciate your group for working so hard and  collaboratively to address the privacy risks that PING identified, and integrating mitigations into the default behavior in the spec so that Web users and sites can benefit from this new, exciting functionality without putting users at risk. Congratulations to you all again

**Comment by anssiko** [comment]
@kenchris @arskama in preparation for our TPAC discussion https://github.com/w3c/devicesensors-wg/issues/69, can you please triage the remaining open issues and:
- label any [open issues](https://github.com/w3c/compute-pressure/issues) considered out of scope for the expected CR, including any new features, as https://github.com/w3c/compute-pressure/labels/V2
- and then, identify issues out of the [remaining issues that are not V2](https://github.com/w3c/compute-pressure/issues?q=is%3Aissue+is%3Aopen+-label%3AV2) that are substantive bug fixes to be addressed prior CR, and label them as https://github.com/w3c/compute-pressure/labels/bug

You can also use [other labels](https://github.com/w3c/compute-pressure/labels) as appropriate, but check the assignment for those two (V2, bug) are up to date.

This will help the group make an informed decision on the publication readiness. Thank you!

**Comment by arskama** [comment]
https://github.com/w3c/compute-pressure/labels/V2 labels:
<ol>
<li> v2 labels under investigation, higher prirority, or requires discussions: </li>
  <ul>
    <li> #8 : Working on a solution, requires some discussion with isv's and chromium team. </li>
    <li> #233: Buffering of CP statuses before requesting: Subject to discussion </li> 
  </ul>
</li>
<li> v2 labels out of current scope: </li>
  <ul>
    <li>#9 Specification wise easy, but implentation is complex. no OS APIs or paltform API available. </li>
   <li>#119: No investigation on metrics and algorithm. </li>
   <li>#120: Requires discussion on possible usage, metrics and memory types, platform and OS API available. wide area of research.
   </li>
   <li>#228: Catch-All pressure: Not feasible at the moment, very complex to define. </li>
   <li>#246:  Not investigated. </li>
   <li> #249:  Not invetigated. </li>
    <li>#288:  Not investigated. </li>
  </ul>
</ol>
<br>

https://github.com/w3c/compute-pressure/labels/bug labels:
#243: No easy fix before CR.
#281: Could be fixed before CR.
#291: Could be fixed before CR.

**Comment by anssiko** [comment]
Thanks @arskama. If there are bugs that you think should be out of scope for CR, you could use a combination of https://github.com/w3c/compute-pressure/labels/bug + https://github.com/w3c/compute-pressure/labels/V2 for those.

**Comment by himorin** [comment]
also I believe 'no', but is there any (additional) at-risk feature we should mark?

**Comment by anssiko** [comment]
@himorin I believe [thermals](https://www.w3.org/TR/compute-pressure/#dom-pressuresource-thermals) pressure source could be marked as [at risk](https://www.w3.org/policies/process/#at-risk).

**Comment by anssiko** [comment]
Current status:

- The only at risk feature identified has been removed in https://github.com/w3c/compute-pressure/pull/300 
- The only remaining https://github.com/w3c/compute-pressure/labels/bug https://github.com/w3c/compute-pressure/issues/243 has a PR #303 in review.
- The rest of the open issues are either deferrer to https://github.com/w3c/compute-pressure/labels/V2 or considered non-blocking for CR based on [our TPAC discussion](https://www.w3.org/2024/09/24-dap-minutes.html#t26).

@himorin I plan to initiate a CfC to publish the CR when PR #303 has landed. Please let us know if you identify any further areas that require attention prior to the expected publication.

**Comment by anssiko** [comment]
[CfC to publish Compute Pressure API Candidate Recommendation Snapshot - review by 29 Nov 2024](https://lists.w3.org/Archives/Public/public-device-apis/2024Nov/0007.html)

**Comment by anssiko** [comment]
To summarize the improvements made in response to the wide review and Origin Trial feedback, please see all the changes since this wide review was initiated:

- [HTML diff](https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fwww.w3.org%2FTR%2F2023%2FWD-compute-pressure-20230203%2F&doc2=https%3A%2F%2Fwww.w3.org%2FTR%2Fcompute-pressure%2F) since 2023-02-03 until the latest
- [commits](https://github.com/w3c/compute-pressure/commits/main/?since=2023-02-03&until=2024-12-05) since 2023-02-03 until 2024-12-05

These changes were made to:
- satisfy wide review feedback as documented and referenced in this issue (initiated 2023-02-07 completed 2023-12-13)
- address feedback from the Origin Trial in [Chrome 115 to 123](https://developer.chrome.com/origintrials/#/view_trial/1196831600973709313) (started 2023-07-20 ended 2024-04-02)
- improve specification clarity informed by implementation experience

The API shipped in Chrome 125 on 2024-05-28.

Looking at the timelines, we observe both the wide review and Origin Trial were partially run in parallel. To ensure specification stability during this overlapping period, while the Origin Trial was active only a small number minor changes not attributable to the wide review feedback were made to the specification, see [commits since 2023-07-20 until 2024-04-02](https://github.com/w3c/compute-pressure/commits/main/?since=2023-07-20&until=2024-04-02). Based on my assessment, the changes during this period do not warrant re-review.

@kenchris please let @himorin know of any substantial changes since 2023-02-03 until 2024-12-05 that might benefit from re-review. Considering changes during OT were minimized, I'd focus on [commits since 2024-04-02 until 2024-12-05](https://github.com/w3c/compute-pressure/commits/main/?since=2024-04-02&until=2024-12-05). Changes that have no user or script-visible impact are usually out of scope for such re-reviews, while any new features would be of interest. Please also review PR https://github.com/w3c/compute-pressure/pull/306 that documents substantive changes since the First Public Working Draft publication 2022-12-20. Your contributions will help advance with the CR transition request https://github.com/w3c/transitions/issues/681.

**Comment by kenchris** [comment]
We made a few renames, but they don't warrant re-review, and our privacy fixes have been carefully reviewed and approved by the PING. We added an accessibility section, but that was also discussed and contiributed by the APA WG.
